### ISSUE TITLE
With .Net 5 Preview8,  VSCode loading projects failed with error Microsoft.Build.Exceptions.InvalidProjectFileException: Invalid static method invocation syntax.

### ISSUE THREAD

**Issue by dandanmu** [OmniSharp/omnisharp-vscode#3971]
## Environment data
`dotnet --info` output:
.NET SDK (reflecting any global.json):
 Version:   5.0.100-preview.8.20406.14
 Commit:    3c12184a70

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.100-preview.8.20406.14/

Host (useful for support):
  Version: 5.0.0-preview.8.20381.10
  Commit:  7b82cf1105

.NET SDKs installed:
  5.0.100-preview.8.20406.14 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.0-preview.8.20404.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.0-preview.8.20381.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

VS Code version:1.47.3
C# Extension version:1.23.0

## OmniSharp log
```
OmniSharp server started with Mono 6.12.0.
    Path: /Users/e2euser/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.0/omnisharp/OmniSharp.exe
    PID: 865

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.14.6 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.6.0 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.6.0 - "/Users/e2euser/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.0/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.6.0 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /Users/e2euser/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.0/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/e2euser/test1/cs1'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Users/e2euser/test1/cs1'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/e2euser/test1/cs1/cs1.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/e2euser/test1/cs1'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/e2euser/test1/cs1/cs1.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/e2euser/test1/cs1' on host 848.
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/Users/e2euser/test1/cs1/cs1.csproj'.
/Users/e2euser/test1/cs1/cs1.csproj(1,1)
Microsoft.Build.Exceptions.InvalidProjectFileException: Invalid static method invocation syntax: "[MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')". Method '[MSBuild]::GetTargetFrameworkIdentifier' not found. Static method invocation should be of the form: $([FullTypeName]::Method()), e.g. $([System.IO.Path]::Combine(`a`, `b`)). Check that all parameters are defined, are of the correct type, and are specified in the right order.  /usr/local/share/dotnet/sdk/5.0.100-preview.8.20406.14/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets
  at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject (System.String errorSubCategoryResourceName, Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, System.Object[] args) [0x00040] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Shared.ProjectErrorUtilities.VerifyThrowInvalidProject[T1,T2] (System.Boolean condition, System.String errorSubCategoryResourceName, Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, T1 arg0, T2 arg1) [0x00003] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject[T1,T2] (Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, T1 arg0, T2 arg1) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Expander`2+Function`1[P,I,T].Execute (System.Object objectInstance, Microsoft.Build.Evaluation.IPropertyProvider`1[T] properties, Microsoft.Build.Evaluation.ExpanderOptions options, Microsoft.Build.Shared.IElementLocation elementLocation) [0x003f9] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Expander`2+PropertyExpander`1[P,I,T].ExpandPropertyBody (System.String propertyBody, System.Object propertyValue, Microsoft.Build.Evaluation.IPropertyProvider`1[T] properties, Microsoft.Build.Evaluation.ExpanderOptions options, Microsoft.Build.Shared.IElementLocation elementLocation, Microsoft.Build.Evaluation.UsedUninitializedProperties usedUninitializedProperties, Microsoft.Build.Shared.FileSystem.IFileSystem fileSystem) [0x0012e] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Expander`2+PropertyExpander`1[P,I,T].ExpandPropertiesLeaveTypedAndEscaped (System.String expression, Microsoft.Build.Evaluation.IPropertyProvider`1[T] properties, Microsoft.Build.Evaluation.ExpanderOptions options, Microsoft.Build.Shared.IElementLocation elementLocation, Microsoft.Build.Evaluation.UsedUninitializedProperties usedUninitializedProperties, Microsoft.Build.Shared.FileSystem.IFileSystem fileSystem) [0x0017f] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Expander`2+PropertyExpander`1[P,I,T].ExpandPropertiesLeaveEscaped (System.String expression, Microsoft.Build.Evaluation.IPropertyProvider`1[T] properties, Microsoft.Build.Evaluation.ExpanderOptions options, Microsoft.Build.Shared.IElementLocation elementLocation, Microsoft.Build.Evaluation.UsedUninitializedProperties usedUninitializedProperties, Microsoft.Build.Shared.FileSystem.IFileSystem fileSystem) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Expander`2[P,I].ExpandIntoStringLeaveEscaped (System.String expression, Microsoft.Build.Evaluation.ExpanderOptions options, Microsoft.Build.Shared.IElementLocation elementLocation) [0x00028] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluatePropertyElement (Microsoft.Build.Construction.ProjectPropertyElement propertyElement) [0x0009c] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluatePropertyGroupElement (Microsoft.Build.Construction.ProjectPropertyGroupElement propertyGroupElement) [0x0002e] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].PerformDepthFirstPass (Microsoft.Build.Construction.ProjectRootElement currentProjectOrImport) [0x0012e] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluateImportElement (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement) [0x00040] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].PerformDepthFirstPass (Microsoft.Build.Construction.ProjectRootElement currentProjectOrImport) [0x00209] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluateImportElement (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement) [0x00040] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].PerformDepthFirstPass (Microsoft.Build.Construction.ProjectRootElement currentProjectOrImport) [0x00209] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluateImportElement (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement) [0x00040] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].PerformDepthFirstPass (Microsoft.Build.Construction.ProjectRootElement currentProjectOrImport) [0x002cc] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].Evaluate () [0x0013c] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].Evaluate (Microsoft.Build.Evaluation.IEvaluatorData`4[P,I,M,D] data, Microsoft.Build.Construction.ProjectRootElement root, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, System.Int32 maxNodeCount, Microsoft.Build.Collections.PropertyDictionary`1[T] environmentProperties, Microsoft.Build.BackEnd.Logging.ILoggingService loggingService, Microsoft.Build.Evaluation.IItemFactory`2[S,T] itemFactory, Microsoft.Build.Evaluation.IToolsetProvider toolsetProvider, Microsoft.Build.Evaluation.ProjectRootElementCacheBase projectRootElementCache, Microsoft.Build.Framework.BuildEventContext buildEventContext, Microsoft.Build.BackEnd.SdkResolution.ISdkResolverService sdkResolverService, System.Int32 submissionId, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext, System.Boolean interactive) [0x00048] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.Reevaluate (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00063] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00035] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00007] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.Initialize (System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x0012b] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00075] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.ProjectCollection.LoadProject (System.String fileName, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion) [0x000f7] in <39698004d9434b949702fb9917b62bee>:0 
  at Microsoft.Build.Evaluation.ProjectCollection.LoadProject (System.String fileName, System.String toolsVersion) [0x00000] in <39698004d9434b949702fb9917b62bee>:0 
  at OmniSharp.MSBuild.ProjectLoader.EvaluateProjectFileCore (System.String filePath) [0x0003f] in <40b5cb7bb6fc439596b63c11039a3732>:0 
  at OmniSharp.MSBuild.ProjectLoader.BuildProject (System.String filePath) [0x0000d] in <40b5cb7bb6fc439596b63c11039a3732>:0 
  at OmniSharp.MSBuild.ProjectFile.ProjectFileInfo.Load (System.String filePath, OmniSharp.MSBuild.ProjectIdInfo projectIdInfo, OmniSharp.MSBuild.ProjectLoader loader, System.Guid sessionId, OmniSharp.Services.DotNetInfo dotNetInfo) [0x00015] in <40b5cb7bb6fc439596b63c11039a3732>:0 
  at OmniSharp.MSBuild.ProjectManager+<>c__DisplayClass31_0.<LoadProject>b__0 () [0x00000] in <40b5cb7bb6fc439596b63c11039a3732>:0 
  at (wrapper delegate-invoke) System.Func`1[System.ValueTuple`3[OmniSharp.MSBuild.ProjectFile.ProjectFileInfo,System.Collections.Immutable.ImmutableArray`1[OmniSharp.MSBuild.Logging.MSBuildDiagnostic],OmniSharp.MSBuild.Notification.ProjectLoadedEventArgs]].invoke_TResult()
  at OmniSharp.MSBuild.ProjectManager.LoadOrReloadProject (System.String projectFilePath, System.Func`1[TResult] loader) [0x0001b] in <40b5cb7bb6fc439596b63c11039a3732>:0 

[fail]: OmniSharp.MSBuild.ProjectManager
        Attempted to update project that is not loaded: /Users/e2euser/test1/cs1/cs1.csproj
```

## Steps to reproduce
1.Install VSCode - https://update.code.visualstudio.com/1.47.3/darwin/stable.
2.Launch VSCode and install the C# extension - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.23.0
3.Open the project in VSCode.

## Expected  behavior
The project should load succeeded.

## Actual behavior
The project load failed.
![image](https://user-images.githubusercontent.com/65638819/89988298-eb1ba100-dcb1-11ea-8a79-49c6e41178d3.png)

##Note:
We tried below steps, but the issue is still repro.
step1.Install Mono 6.12.0.90 from https://www.mono-project.com/download/preview/ and set “omnisharp.useGlobalMono” to "auto".
step2.Enable “LegacySdkResolver”  by creating omnisharp.json as described below, in ~/.omnisharp/ or project directory.
{
   "msbuild": {
       "UseLegacySdkResolver": true
   }
}

**Comment by JoeRobich** [comment]
@dandanmu Merge a change into OmniSharp Roslyn that may address this issue. If you set your "omnisharp.path" to "latest" and restart. Does this project load error go away?

**Comment by dandanmu** [comment]
We have tried to install Mono6.12.0.90, enable “LegacySdkResolver” and set "omnisharp.path" to "latest" and restart, but the project was still load failed.

**Comment by JoeRobich** [comment]
@dandanmu Sorry that my recommendation keeps changing. If you set "omnisharp.useGlobalMono" to "never" are you able to load your project?

**Comment by dandanmu** [comment]
Thanks, the project can load successfully after setting “omnisharp.useGlobalMono” to "never".

**Comment by JoeRobich** [comment]
@dandanmu Ok, so it took updating target files in the OmniSharp MSBuild install.

With the lastest 1.23.0 C# extension installed.
- Set "omnisharp.path" to "latest" (To pull in the correct msbuild target files)
- Set "omnisharp.useGlobalMono" to "never"
- No need for "UseLegacySdkResolver" in omnisharp but shouldn't hurt

**Comment by jmarolf** [comment]
fixed in version 1.23.1
